File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        String answer = "";
		        int a = 0;
		        StringBuilder sb = new StringBuilder();
		        for(int i = 1; i<numLog.length; i++){
		            a = numLog[i] - numLog[i-1];
		            if(a == 1){
		                sb.append("w");
		            }else if(a == -1){
		                sb.append("s");
		            }else if(a == 10){
		                sb.append("d");
		            }else if(a == -10){
		                sb.append("a");
		            }
		        }
		        answer = sb.toString();
		        return answer;
    }
}

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    answer = ''
    
    for i in range(len(numLog) - 1):
        number = numLog[i + 1] - numLog[i]
        
        if number == 1:
            answer += 'w'
        elif number == -1:
            answer += 's'
        elif number == 10:
            answer += 'd'
        elif number == -10:
            answer += 'a'
    
    return answer